packages/fxa-settings/src/lib/gql.test.ts

```diff
@@ -0,0 +1,143 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { errorHandler } from './gql';
+import { ErrorResponse } from '@apollo/client/link/error';
+import { Operation, NextLink } from '@apollo/client/core';
+import { GraphQLError } from 'graphql';
+import { mock } from 'jest-mock-extended';
+
+let errorResponse: ErrorResponse;
+
+describe('errorHandler', () => {
+  let realLocation: Location;
+
+  beforeAll(() => {
+    realLocation = window.location;
+
+    // We don't verify that console.error gets called, but mocking it out
+    // avoids console noise filling up the command line as the tests run.
+    console.error = jest.fn();
+  });
+
+  afterAll(() => {
+    window.location = realLocation;
+    console.error.mockRestore();
+  });
+
+  // Note: it's surprisingly difficult to mock out window.location cleanly.
+  // Messing with globals is gross, but works, and is really simple.
+  function setLocation({ search, pathname }) {
+    delete window.location;
+    window.location = {
+      replace: jest.fn(),
+      search,
+      pathname,
+    };
+  }
+
+  describe('if called with a GQL authentication error', () => {
+    beforeEach(() => {
+      errorResponse = {
+        graphQLErrors: [
+          new GraphQLError(
+            'Incorrect password',
+            undefined,
+            undefined,
+            undefined,
+            undefined,
+            undefined,
```

Comment:
I'm not a fan of this explicit passing of `undefined` thing. suggestions welcome on a nicer approach :-)